client/src/main/scala/com/dwolla/cloudflare/domain/model/ratelimits/package.scala

```diff
@@ -0,0 +1,173 @@
+package com.dwolla.cloudflare.domain.model
+
+import com.dwolla.circe._
+import io.circe.{Decoder, Encoder}
+import shapeless.tag.@@
+
+package object ratelimits {
+
+  type RateLimitId = String @@ RateLimitIdTag
+
+  private[cloudflare] val tagRateLimitId: String ⇒ RateLimitId = shapeless.tag[RateLimitIdTag][String]
+
+}
+
+package ratelimits {
```

Comment:
Moving all these classes into this file makes this PR a lot harder to review, especially since most of this stuff doesn't need to be in the package object, because it's more difficult to understand what's actually changed and what has just been moved around. It's probably fine to leave it alone for now, but in the future I'd prefer to see this kind of refactoring as a separate PR (one for moving things around without changing behavior, and then one for changing the behavior) so it's easier to figure out what's going on.

Reply:
My apologies for the large change set. You make a valid point. Much of the original domain objects were not accurate, so I figured as part of my changes I'd update what existed to emulate the most recent PageRule endpoint addition where there is a simplified domain model without DTOs.

Out of curiosity, could you elaborate on why some of what I added would be more appropriate outside of a package object (or I guess isn't required to be in a package object)?

Reply:
I just meant that everything in the `package ratelimits { … }` block (which is the equivalent of putting `package ratelimits` at the top of the file, just slightly different syntax to make the package decl apply to a different amount of the file) could be moved to separate files (within the limitations of `sealed`, etc.). When I was reviewing this the other day, it looked like a lot of things had been moved from separate files (`RateLimit.scala`, `RateLimitAction.scala`, etc.) that strictly speaking didn't need to be moved to compile. Whether those things should have been spread out over multiple files or consolidated is more a matter of style (I'm on board with consolidating here), but leaving them in place while we updated those type definitions calls attention to the actual changes being made.

The stuff that is in the actual `package object ratelimits { … }` is fine—that stuff all needs to be in the package object to compile.

Reply:
Gotcha. I see what you mean. Thanks for the clarification.